core/arch/arm/pta/sdp_pta.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2017, ARM Limited, All Rights Reserved
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <kernel/pseudo_ta.h>
+#include <mm/core_memprot.h>
+#include <mm/tee_mmu.h>
+#include <sdp_pa.h>
+
+#define PTA_NAME "sdp.pta"
+
+static TEE_Result sdp_pa_cmd_virt_to_phys(uint32_t types,
+					  TEE_Param params[TEE_NUM_PARAMS])
+{
+	char *va = params[0].memref.buffer;
+	size_t len = params[0].memref.size;
+	paddr_t pa;
+	struct tee_ta_session *s;
+	struct user_ta_ctx *utc;
+	TEE_Result res;
+
+	if (types != TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+				     TEE_PARAM_TYPE_VALUE_OUTPUT,
+				     TEE_PARAM_TYPE_NONE,
+				     TEE_PARAM_TYPE_NONE)) {
+		EMSG("bad parameters types: %x", (unsigned int)types);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	s = tee_ta_get_calling_session();
+	if (!s)
+		return TEE_ERROR_ACCESS_DENIED;
+
+	utc = to_user_ta_ctx(s->ctx);
+	res = tee_mmu_check_access_rights(utc, TEE_MEMORY_ACCESS_READ |
```

Comment:
Sorry, nitpicking, but I wonder if this is useful. When a TA invoke a pTA, its arguments are verified, and I think memref parameters are already validated against any owner access (especially since TAs are not allowed to expose their private memory to other TAs).

This can be later fix as an optimization...

Reply:
Ok.

Reply:
TBD?

Reply:
I think it is useful to check the memory access right for TA. It prevent the case that TA maybe have not right to access some SDP address. Of course, TA also need check the memref parameters. But it may assign another address to memref parameters which not belong it. 
If you do not think that, please add comment:)